edit-post/components/header/plugins/index.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import { map, isEmpty, isString } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { NavigableMenu, withInstanceId, IconButton, MenuItemsSeparator } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { getEditorMenuItems } from '../../../api';
+
+/**
+ * Renders a list of plugins that will activate different UI elements.
+ *
+ * @param   {Object} props The component props.
+ * @return {Object} The rendered list of menu items.
+ */
+function Plugins( props ) {
+	const ellipsisMenuItems = getEditorMenuItems();
+
+	if ( isEmpty( ellipsisMenuItems ) ) {
+		return null;
+	}
+
+	const { instanceId } = props;
+
+	/**
+	 * Handles the user clicking on one of the plugins in the menu
+	 *
+	 * Does nothing currently, but should be used to trigger the plugins sidebar
+	 *
+	 * @param {string} pluginId The plugin id.
+	 * @return {void}
+	 */
+	function onSelect( pluginId ) {
+		props.onSelect();
+		ellipsisMenuItems[ pluginId ].callback();
+	}
+
+	const labelId = `components-choice-menu-plugins-label-${ instanceId }`;
+
+	return [
+		<MenuItemsSeparator key="plugins-separator" />,
```

Comment:
Is there anything that prevents us from using `MenuItemsGroup` here? With the refactoring done in #5161 you no longer need `MenuItemSeparator`. It is also possible to use hooks to dynamically inject menu items to such group:

```js
<MenuItemsGroup
	label={ __( 'Plugins' ) }
	filterName="editPost.MoreMenu.plugins"
/>
```

and you can add new menu items there as described in my comment: https://github.com/WordPress/gutenberg/pull/4484#issuecomment-367593808.